File: sort_algorithms/insertion_sort.py
# Insertion Sort
# Python
# Porter L

def insertionSort(Arr): 
    for x in range(1, len(Arr)): 
        key = Arr[x] 
        y = x-1
        while y >= 0 and key < Arr[y] : 
            Arr[y + 1] = Arr[y] 
            y -= 1
        Arr[y + 1] = key 
    return Arr

# Example
if __name__=='__main__': 
    test_array = [2,7,22,71,5,3,11,9,7,5,4,3,56,3,23,43,97,3]
    print(test_array) # unsorted
    test_array = insertionSort(test_array)
    print(test_array) # sorted

File: sort_algorithms/insertion_sort/insertion_sort.php
<?php
// Insertion Sort
// PHP
// Porter L

function insertionSort($Arr)
{
    for($x=0;$x<count($Arr);$x++){
        $num = $Arr[$x];
        $y = $x-1;
        while($y>=0 && $Arr[$y] > $num){
            $Arr[$y+1] = $Arr[$y];
            $y--;
        }
        $Arr[$y+1] = $num;
    }
    return $Arr;
}

// Example
$test_array = array(2,7,22,71,5,3,11,9,7,5,4,3,56,3,23,43,97,3);
echo "Original Array : ";
echo implode(', ',$test_array );
echo "\nSorted Array : ";
echo implode(', ',insertionSort($test_array));
echo "\n";

?>